mt-kahypar/partition/coarsening/deterministic_multilevel_coarsener.cpp

```diff
@@ -184,12 +320,15 @@ size_t DeterministicMultilevelCoarsener<TypeTraits>::approveVerticesInTooHeavyCl
       // could be parallelized without extra memory but factor 2 work overhead and log(n) depth via binary search
       for (; ; ++first_rejected) {
         // we know that this cluster is too heavy, so the loop will terminate before
-        assert(first_rejected < nodes_in_too_heavy_clusters.size());
-        assert(propositions[nodes_in_too_heavy_clusters[first_rejected]] == target);
+        ASSERT(first_rejected < nodes_in_too_heavy_clusters.size());
+        ASSERT(propositions[nodes_in_too_heavy_clusters[first_rejected]] == target);
         HypernodeID v = nodes_in_too_heavy_clusters[first_rejected];
         if (target_weight + hg.nodeWeight(v) > _context.coarsening.max_allowed_node_weight) {
           break;
         }
+        if (has_fixed_vertices && !fixed_vertices.contract(target, v)) {
```

Comment:
Do we have to fix the opportunistic_cluster_weight of v in this case? Or is it fine since we don't decrement in the first place? Do we even have to check fixed vertex policy again here? It was checked in `calculatePreferredTargetCluster`